### PR TITLE
update references for v0.18

### PR DESCRIPTION
**Pull request by kalantar**
- updated extension documentation because of code changes
- updated source references to mostly be v17 or v18 instead of older even if not strictly necessary

### CODE REVIEW

File: docs/tutorials/integrations/kserve/canary.md [hunk 1]
```diff
@@ -225,7 +225,7 @@ kubectl delete isvc/wisdom-1
 Delete routing:
 
 ```shell
-cat <<EOF | helm template routing --repo https://iter8-tools.github.io/iter8 routing-actions --version 0.1.5 -f - | kubectl delete -f -
+cat <<EOF | helm template routing --repo https://iter8-tools.github.io/iter8 routing-actions --version 0.18 -f - | kubectl delete -f -
```
Comment (Alan-Cha) [inline docs/tutorials/integrations/kserve/canary.md:228]: So we decided to have `routing-actions` match the minor version of Iter8?

File: docs/tutorials/load-test-grpc-multiple.md [hunk 2]
```diff
@@ -32,17 +32,17 @@ template: main.html
...
+--repo https://iter8-tools.github.io/iter8 --version 0.18 routeguide-test iter8 \
 --set "tasks={ready,grpc}" \
 --set ready.deploy=routeguide \
 --set ready.service=routeguide \
 --set ready.timeout=60s \
 --set grpc.host=routeguide.default:50051 \
 --set grpc.protoURL=https://raw.githubusercontent.com/grpc/grpc-go/v1.52.0/examples/route_guide/routeguide/route_guide.proto \
 --set grpc.endpoints.getFeature.call=routeguide.RouteGuide.GetFeature \
---set grpc.endpoints.getFeature.dataURL=https://raw.githubusercontent.com/iter8-tools/docs/v0.13.13/samples/grpc-payload/unary.json \
+--set grpc.endpoints.getFeature.dataURL=https://raw.githubusercontent.com/iter8-tools/docs/v0.17.3/samples/grpc-payload/unary.json \
```
Comment (Alan-Cha) [inline docs/tutorials/load-test-grpc-multiple.md:43]: Is this intended to be `17.3` or `18.3`? Does this need to change? I thought we agreed that links to the docs don't need to change

File: docs/getting-started/first-abn.md [hunk 3]
```diff
@@ -31,25 +31,23 @@ A sample application using the Iter8 SDK is provided. Deploy both the frontend a
 
     === "node"
         ```shell
-        kubectl create deployment frontend --image=iter8/abn-sample-frontend-node:0.15.0
+        kubectl create deployment frontend --image=iter8/abn-sample-frontend-node:0.17.3
```
Comment (Alan-Cha) [inline docs/getting-started/first-abn.md:34]: Should these be 18.3?

**Comment by Alan-Cha** [comment]
I find it confusing to update some to 17 and some to 18. I'm worried this may lead to a mistake.

File: docs/tutorials/integrations/kserve/canary.md (hunk 1 above)
Comment (kalantar) [reply docs/tutorials/integrations/kserve/canary.md:228]: Yes, this is what @sriumcp also recommended.

File: docs/tutorials/load-test-grpc-multiple.md (hunk 2 above)
Comment (kalantar) [reply docs/tutorials/load-test-grpc-multiple.md:43]: These links refer to the docs repo. v0.18 has not yet been released so reference to 0.17 works.  References to the iter8 repo are updated to v0.18.

File: docs/getting-started/first-abn.md (hunk 3 above)
Comment (kalantar) [reply docs/getting-started/first-abn.md:34]: No, this is in docs repo so 17 still makes sense.

**Review by Alan-Cha — APPROVED** [review]

**Pull request merged**